src/zulip/api.rs

```diff
@@ -228,9 +243,13 @@ impl ZulipApi {
         let add_ids = serialize_as_array(add_ids);
         submit(reqwest::Method::POST, subscriptions, add_ids)?;
 
-        let subscriptions = format!("[{stream_name}]");
-        let remove_ids = serialize_as_array(remove_ids);
-        submit(reqwest::Method::DELETE, subscriptions, remove_ids)
+        if !self.is_stream_public(stream_id)? {
```

Comment:
We probably want to do this at the diff level, otherwise for something like t-compiler we'd generate huge diffs every time.

Reply:
Fixed!